### ISSUE TITLE
Opening a folder on NFS fails with OSError [OSX]

### ISSUE THREAD

**Issue by ehlertjd** [PyFilesystem/pyfilesystem2#217]
## Steps to Reproduce
- Have an NFS share mounted at `/nfs`
- Attempt to create an OSFS object with that directory:
```python
from fs.osfs import OSFS
nfs = OSFS('/nfs')
```
## Expected result
NFS mounted folders should be treated as if they were normal folders.

## Actual result
Get an OSError when trying to get `max_sys_path_length`:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/fs/osfs.py", line 146, in __init__
    fsencode(_root_path), os.pathconf_names["PC_PATH_MAX"]
OSError: [Errno 22] Invalid argument
```

## Workaround
Open the root folder, then open the NFS shared folder as a subdirectory:
```python
from fs.osfs import OSFS
rootfs = OSFS('/')
nfs = rootfs.opendir('/nfs')
```

**Comment by willmcgugan** [comment]
Just had a look at this one, and I was unable to even get nfs mounted on my Macbook.

From the looks of things, I could just catch that exception and carry on. The side-effect would be that OSFS wouldn't be able to detect and validate the maximum path size.

@ehlertjd Would you be able to test that for me if I make that change?

**Comment by ehlertjd** [comment]
@willmcgugan Absolutely! Although it may not be until early next week.

**Comment by willmcgugan** [comment]
That's great, thanks. I probably won't be able to get to it until next week.

**Comment by willmcgugan** [comment]
This issue *should* be fixed in 2.4.4. 

Hopefully this is not hubris on my part, because I never reproduced it. I'm assuming that nfs doesn't expose the maximum path size for whatever reason. 

Feel free to close the issue if it works for you!

**Comment by ehlertjd** [comment]
Confirmed that this was fixed in 2.4.4
